src/study_api.jl

```diff
@@ -142,12 +270,24 @@ function get_vector(
         error("Invalid attribute '$attr' for object of type '$name'")
     end
 
-    vector = element[attr]
+    return element[attr]::Vector
+end
+
+function get_vector(
+    data::Data,
+    name::String,
+    index::Integer,
+    attr::String,
+    type::Union{Type{<:T},Nothing}=nothing,
+) where {T<:MainTypes}
+    vector = _get_vector_ref(data, name, index, attr)
+
+    attr_data = _get_attr_data(data, name, attr)
 
     if isnothing(type)
-        return Vector(vector)
+        return _parse_vector(vector, attr_data.type)
     else
-        return Vector{type}(vector)
+        return _parse_vector(vector, type)
     end
```

Comment:
Should it be possible to parse with a different type?

Reply:
Here I was just following existent the API where the user tells which type he wants entries to be.
The question is: can we trust that `attr_data` will always have the correct type?
Also, I just came to this problem where `::Nothing` is not in `MainTypes` while there is `null` everywhere in JSON data.